qt/python/mantidqt/widgets/algorithmprogress/dialog_presenter.py

```diff
@@ -13,9 +13,13 @@ class AlgorithmProgressDialogPresenter(AlgorithmProgressPresenterBase):
     """
     Presents progress reports on algorithms.
     """
+
     def __init__(self, view, model):
         super(AlgorithmProgressDialogPresenter, self).__init__()
-        view.close_button.clicked.connect(self.close)
+        # connect the close button to the closeEvent of the window
+        # so that pressing the X button, and pressing `Close` go through
+        # the same routine, and properly call the presenter's close()
+        view.close_button.clicked.connect(view.closeEvent)
```

Comment:
The connection should be to `view.close` which will in turn call `closeEvent(event)` passing in a proper `QCloseEvent`.